jekyll/_cci2/docker-hub-pull-through-mirror.adoc

```diff
@@ -20,60 +20,135 @@ Starting https://www.docker.com/blog/scaling-docker-to-serve-millions-more-devel
+=== Set up a secure pull through cache registry (HTTPS proxy for Docker Hub)
+
+Under certain circumstances (especially when you are enabling https://docs.docker.com/develop/develop-images/build_enhancements/[BuildKit] on machine executors and remote docker engines), your pull through cache registry needs to accept connections over HTTPS. This is the instruction to setup a pull through cache that listens for HTTPS connections.
 
-1. Run the command below to *configure the `registry-mirrors` option for the Docker daemon*.
+. *Set up a Docker Hub account* which will be used by the pull through cache registry to access Docker Hub.
 +
-NOTE: Replace `192.0.2.1` with the IP address of your pull through cache server.
+TIP: Setting up a paid account can be a good option to ease the rate limiting further. See also: https://www.docker.com/pricing
+
+. *Set up an independent Linux server* that has Docker installed.
 +
-`sudo bash -c 'cat <<< $(jq ".\"registry-mirrors\" = [\"http://192.0.2.1\"]" /etc/docker/daemon.json) > /etc/docker/daemon.json'`
+We set up this registry as an independent server (i.e. outside of CircleCI boxes) to avoid load on this cache registry affecting other CircleCI Server services.
++
+Note that **the sever needs to be accessible by its hostname**. Namely, you have to configure DNS accordingly so that the hostname (assume `your-mirror.example.com`) resolves to the IP address of the server correctly. Hereby we assume that the URL for the registry to setup is `https://your-mirror.example.com` (be aware that the URL scheme is `http**_s_**`, not `http`). The URL will be needed later to arm Nomad clients and the VM Service.
 
-2. *Reload Docker daemon* to apply the configuration.
+. *Obtain a TLS certificate for `your-mirror.example.com`* and *put the certificate and the private key into `/root/tls`* of the server.
++
+Note that this document assumes that you are obtaining a certificate issued by a well-known CA, not a self-signed one. Use of self-signed certificates is NOT tested.
+
+. Run the command below to *start the registry*.
++
+[source,bash]
+----
+sudo docker run \
+  -d \
+  -p 443:5000 \
+  --restart=always \
+  --name=through-cache-secure \
+  -v /root/tls:/data/tls \
+  -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" \
+  -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME \
+  -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/data/tls/fullchain.pem \
+  -e REGISTRY_HTTP_TLS_KEY=/data/tls/privkey.pem \
+  registry
+----
++
+NOTE:
++
+.. Replace `DOCKER_HUB_USERNAME` and `DOCKER_HUB_ACCESS_TOKEN` with the username for the Docker Hub account and the access token you obtained through https://hub.docker.com/settings/security, respectively.
+.. Replace `fullchain.pem` and `privkey.pem` with the filenames of your certificate and private key, respectively. *The certificate specified there must be correctly chained enough for tracing the certification path from the leaf to the root*.
+
```

Comment:
Could we move these up before the command snippet, just so they don't get missed? Can use a multiline note I think:
```
[NOTE]
====
Replace... content here
====
```